site/content/overview/resource-validation.md

```diff
@@ -141,3 +145,62 @@ Status:
 ```
 
 {{< note>}}Make sure the reported observed generation is the same as the resource generation.{{< /note >}}
+
+## NGINX Gateway Fabric Resource Validation
+
+### Step 1 - OpenAPI Scheme validation by Kubernetes API Server
+
+The Kubernetes API server validates NGINX Gateway Fabric resources against the OpenAPI schema embedded in the NGINX Gateway Fabric CRDs. For example, if you create an NginxGateway with an invalid logging level, "some-level", the API server will reject it with the following error:
+
+```shell
+kubectl apply -f nginx-gateway-config.yaml
+```
+
+```text
+The NginxGateway "nginx-gateway-config" is invalid: spec.logging.level: Unsupported value: "some-level": supported values: "info", "debug", "error"
+```
+
+{{< note >}}While unlikely, bypassing this validation step is possible if the NGINX Gateway Fabric CRDs are modified to remove the validation. If this happens, Step 2 will report an error in the resource's status.{{< /note >}}
+
+### Step 2 - Validation by NGINX Gateway Fabric
+
+This step validates the settings in the NGINX Gateway Fabric CRDs and rejects invalid resources. The validation error is reported via the status and as an event. For example:
```

Comment:
would it make sense to capitalize because Event is a resource?
```suggestion
This step validates the settings in the NGINX Gateway Fabric CRDs and rejects invalid resources. The validation error is reported via the status and as an Event. For example:
```